development

sergiynikon opened the pull request.

sergiynikon committed:
move projects to parent folder and working with code style

sergiynikon committed:
fix issues, created by Bohdatn

sergiynikon committed:
Remake entities and add migration

and add summary to ValuesController

sergiynikon committed:
Remove unnecessary attributes in entities

sergiynikon committed:
Remove unnecessary controllers and add authtest controller

sergiynikon committed:
Drop all migration and create new

also edit migraton, set not nullable to data columns

sergiynikon committed:
Remove unnecessary methods and add usercontroller with getcurrentuser logic

sergiynikon committed:
move files to implementation folders

sergiynikon committed:
Remove unnecessary file

sergiynikon committed:
Revert "Remove unnecessary file"

This reverts commit 1a25b7aedbb83ff8d9d1e24cd40a5f281399fccb.

sergiynikon committed:
Remove unnecessary file and using

sergiynikon committed:
Rename namespaces to the correct ones

sergiynikon committed:
done

sergiynikon committed:
Merge pull request #6 from sergiynikon/feature-get-user-with-transactions

Feature get user with transactions

sergiynikon committed:
Remake TransactionsDto class and change methods in transactionservice

sergiynikon committed:
Remake database, use fluent api instead of attributes

sergiynikon committed:
Merge pull request #7 from sergiynikon/remake-database

Remake database, use fluent api instead of attributes

sergiynikon committed:
Rename to ViewDto

sergiynikon committed:
Merge pull request #8 from sergiynikon/feature-get-user-with-transactions

Feature get user with transactions

sergiynikon committed:
Remove id = new guid() from constructor

it works without it

BodyDuk commented on BankingApp.Services/Implementation/UserService.cs:
```diff
@@ -0,0 +1,35 @@
+            var usersFromDatabase = _unitOfWork.UserRepository.GetAll();
+            foreach (var user in usersFromDatabase)
+            {
+                users.Add(UserViewDto.ConvertFromUser(user));
```
Зроби для иодельки юзер розширення

BodyDuk commented on BankingApp.Services/Helpers/AuthOptions.cs:
```diff
@@ -0,0 +1,35 @@
+        {
+            return new SymmetricSecurityKey(Encoding.ASCII.GetBytes(Key));
+        }
+
```
??????????

BodyDuk commented on BankingApp.Services/Helpers/AuthOptions.cs:
```diff
@@ -0,0 +1,35 @@
+        private static IConfiguration ConstantsConfig { get; } = SetConfiguration()
+            .GetSection("Constants")
+            .GetSection("ForAuthOptions");
+        private static IConfiguration SetConfiguration()
```
?????

BodyDuk commented on BankingApp.DataTransfer/LoginDTO.cs:
```diff
@@ -0,0 +1,13 @@
+    {
+        [Required]
+        public string Login { get; set; }
+
```
Додати атрибути валідації

BodyDuk commented on BankingApp.DataTransfer/AuthenticationDetailsDto.cs:
```diff
@@ -0,0 +1,31 @@
+                isSuccess: true,
+                message: Constants.SuccessMessage);
+        }
+        
```
??????

BodyDuk commented on BankingApp.Data/Repositories/Implementation/Repository.cs:
```diff
@@ -0,0 +1,56 @@
+    public class Repository<TEntity> : IRepository<TEntity> where TEntity : class
+    {
+        private readonly DbContext _context;
+        protected readonly DbSet<TEntity> Entity;
```
Зроби це проперті

BodyDuk commented on BankingApp.Data/Migrations/20190704141057_Third.cs:
```diff
@@ -0,0 +1,17 @@
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+
```
??????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????

BodyDuk commented on BankingApp.Data/Migrations/20190704135113_Sec.cs:
```diff
@@ -0,0 +1,17 @@
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
```
???????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????

BodyDuk commented on BankingApp.Data/Migrations/20190704134715_Second.cs:
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace BankingApp.Data.Migrations
+{
```
???????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????????

BodyDuk commented on BankingApp.Data/Entities/User.cs:
```diff
@@ -0,0 +1,30 @@
+        public string Password { get; set; }
+        public long Balance { get; set; }
+
+        public virtual ICollection<Transaction> SentTransactions { get; set; } = new List<Transaction>();
```
Провір чи це необхідно

BodyDuk commented on BankingApp.Data/Entities/Transaction.cs:
```diff
@@ -0,0 +1,27 @@
+{
+    public class Transaction
+    {
+
```
?

BodyDuk commented on BankingApp.Data/DataContext.cs:
```diff
@@ -0,0 +1,32 @@
+    {
+        public DataContext(DbContextOptions<DataContext> options) : base(options)
+        {
+        }
```
?

BodyDuk commented on BankingApp.API/Startup.cs:
```diff
@@ -0,0 +1,83 @@
+                    "default",
+                    "{controller=Home}/{action=Index}/{id?}");
+            });
+
```
?

BodyDuk commented on BankingApp.API/Startup.cs:
```diff
@@ -0,0 +1,83 @@
+                options.UseSqlServer(Configuration.GetConnectionString("DefaultConnection"));
+            });
+
+            services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
```
Забери лишнє

sergiynikon commented on BankingApp.API/Controllers/UsersController.cs:
```diff
@@ -0,0 +1,40 @@
+        public IActionResult GetCurrentUser()
+        {
+            var userId = _authenticateService.GetUserId(this.User.Claims);
+
```
done

sergiynikon committed:
Test commit

sergiynikon committed:
Test commit

sergiynikon committed:
Add banking logic

add logic with Deposit, Withdraw and Transfer features

sergiynikon committed:
change token lifetime

sergiynikon committed:
Fix bug closes #11

sergiynikon committed:
Revert "Fix bug closes #11"

This reverts commit 03645937e9975dbf568a074ad01e6cc038b0e2cd.

sergiynikon committed:
Revert "Revert "Fix bug closes #11""

This reverts commit 6f74a9fe9b974dc3cbcc1f57ea9eb4d27217fe91.

sergiynikon committed:
Rewrite .gitignore

sergiynikon committed:
Rewrite .gitignore

sergiynikon committed:
Update .gitignore

sergiynikon committed:
remove Properties\launchSettings.json from github

sergiynikon committed:
Improved code style and deleted old project from github

BodyDuk committed:
Merge pull request #14 from sergiynikon/Work-with-code-style

Work with code style

sergiynikon committed:
Remove unnecessary file

sergiynikon committed:
Add TODOs

sergiynikon committed:
add transaction to specific user when method executed Closes #15 (#16)

sergiynikon committed:
Change return values in controllers

return ResultDto

sergiynikon committed:
Change return value frm AuthenticateDetailsDto to ResultDto

sergiynikon committed:
Merge pull request #12 from sergiynikon/Banking-logic

Banking logic

sergiynikon committed:
Remove AuthenticationDetailsDto

because there already exists ResultDto and AuthenticationDetailsDto is not used now

sergiynikon committed:
Remove empty migrations

sergiynikon committed:
Parallelism problem done

BodyDuk committed:
Merge pull request #20 from sergiynikon/Parallelism-problem

Parallelism problem
